File: LABORATORIOS/LABORATORIO2/Ejercicio10/Java/SecuenciaGenerador.java
import java.util.ArrayList;
import java.util.List;

public class SecuenciaGenerador {
    public static List<Integer> generarSecuencia(int n) {
        List<Integer> secuencia = new ArrayList<>();
        int contador = 0;
        int repeticiones = 1;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < repeticiones; j++) {
                secuencia.add(contador);
            }
            contador = 1 - contador;
            repeticiones++;
        }

        return secuencia.subList(0, n);
    }

    public static void main(String[] args) {
        int nTerminos = 10;
        List<Integer> secuenciaGenerada = generarSecuencia(nTerminos);
        System.out.println(secuenciaGenerada);
    }
}

File: LABORATORIOS/LABORATORIO2/Ejercicio10/Python/SecuenciaGenerador.py
def generar_secuencia(n):
    secuencia = []
    contador = 0
    repeticiones = 1

    for _ in range(n):
        secuencia.extend([contador] * repeticiones)
        contador = 1 - contador
        repeticiones += 1

    return secuencia[:n]

n_terminos = 10
secuencia_generada = generar_secuencia(n_terminos)
print(secuencia_generada)